test/test_nestedtensor.py

```diff
@@ -6984,6 +6999,45 @@ def get_flops(nt):
 
         self.assertEqual(get_flops(nt), get_flops(nt_meta))
 
+    @onlyCUDA
+    def test_device_conversion(self):
+        # cpu -> cuda
+        t_cpu = torch.nested.nested_tensor(
+            [
+                torch.randn(2, 5),
+                torch.randn(3, 5),
+                torch.randn(18, 5),
+            ],
+            layout=torch.jagged,
+            device="cpu",
+        )
+        t_cuda = t_cpu.to("cuda")
+
+        # Shape comparison
+        self.assertEqual(t_cpu.shape, t_cuda.shape)
+
+        # Conversion from host to device and vice versa, discards
+        # the old offsets.
```

Comment:
remind me again why this is the right move here? is it to avoid keeping CUDA offsets alive too long after a conversion to CPU?

Reply:
From the cuda -> cpu direction. I think its slightly less unexpected to no longer hold onto cuda memory. It's also cheap to move from cpu to back to cuda anyway.
The cpu -> cuda direction is the result of (cuda -> cpu getting rid of cuda) in combination with a fakification bug. In the long term, we should keep cpu around when moving to cuda.